### PR TITLE
u2p cli satellite_sync

### PR DESCRIPTION
**Pull request by tstrych**
Please do not merge before @latran will ack this.

Reason of PR: Pytest conversion + f'strings. 

Thinks I am not 100% sure about.
- multiline f-strings. 
- Should I remove classes ? But classes logically differentiate the test content.  If I remove them any proposed solution to keep them divided ? I did use fixtures for this particular reason in `cli/test_host.py`. 

**Before** I start to convert this feature I run the tests. 
The results were: 
```
=========================== short test summary info ============================
FAILED test_satellitesync.py::RepositoryExportTestCase::test_positive_export_rh_product
FAILED test_satellitesync.py::ContentViewSync::test_positive_export_import_default_org_view
FAILED test_satellitesync.py::ContentViewSync::test_positive_export_import_redhat_cv
FAILED test_satellitesync.py::ContentViewSync::test_positive_export_import_redhat_cv_with_huge_contents
===== 4 failed, 17 passed, 41 deselected, 1 warning in 1814.71s (0:30:14) =====
```
Test `test_positive_export_import_default_org_view` did pass on rerun.

I did investigate failiures, problem is with manifest. We do not have required subscriptions for testing. 
So when we want to enable repository from reposet the test fails. Mentioned tests are the only that are trying to enable repo.
And @JacobCallahan already know about the problem, and he made a steps to resolve it.
But till then this will fail. 

**After** 
```
=========================== short test summary info ============================
FAILED test_satellitesync.py::TestRepositoryExport::test_positive_export_rh_product
FAILED test_satellitesync.py::TestContentViewSync::test_positive_export_import_redhat_cv
FAILED test_satellitesync.py::TestContentViewSync::test_positive_export_import_redhat_cv_with_huge_contents
===== 3 failed, 18 passed, 41 deselected, 3 warnings in 1864.33s (0:31:04) =====
```

The time is slightly longer ~1min, but slight change like this can be caused by multiple reasons.

I made this conversion in record time :smile:

### CODE REVIEW

File: tests/foreman/cli/test_satellitesync.py [hunk 1]
```diff
@@ -1006,21 +1106,21 @@ def test_negative_export_cv_with_on_demand_repo(self):
...
             )
-        self.assert_error_msg(
-            error,
-            "Could not export the content view:\n  Error: Ensure the content view version"
-            " '{} {}' has at least one repository.\n".format(cv_name, cv_version),
-        )
+        assert (
+            f"Could not export the content view:\n  "
+            f"Error: Ensure the content view version '{cv_name} {cv_version}' "
+            f"has at least one repository."
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_satellitesync.py:1119]: no f-string needed on the first or third line. single quotes on outer as well

File: tests/foreman/cli/test_satellitesync.py [hunk 2]
```diff
@@ -1049,23 +1149,23 @@ def test_negative_export_cv_with_background_policy_repo(self):
...
             )
-        self.assert_error_msg(
-            error,
-            "Could not export the content view:\n  Error: Ensure the content view version"
-            " '{} {}' has at least one repository.\n".format(cv_name, cv_version),
-        )
+        assert (
+            f"Could not export the content view:\n  "
+            f"Error: Ensure the content view version '{cv_name} {cv_version}' "
+            f"has at least one repository.\n"
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_satellitesync.py:1162]: same thing here and possibly further down

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

**Comment by latran** [comment]
Manifest should be fixed now.  Please add results to test case.

**Comment by tstrych** [comment]
```
============================== short test summary info ==============================
FAILED tests/foreman/cli/test_satellitesync.py::TestContentViewSync::test_positive_export_import_redhat_cv_with_huge_contents
=============== 1 failed, 2 passed, 3 warnings in 3603.91s (1:00:03) ================
```
huge content test timeoted on:
`Package.list({'content-view-version-id': exporting_cvv_id})`

```
paramiko/buffered_pipe.py:160: in read
    self._cv.wait(timeout)
```

**Comment by tstrych** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_satellitesync.py

**Comment by tstrych** [comment]
jenkins for prt is down so prt will fail :/

**Comment by tstrych** [comment]
prt passed :partying_face:

**Review by JacobCallahan — APPROVED** [review]

**Review by mshriver — APPROVED** [review]

**Pull request merged**